### PR TITLE
Adding gcovr code coverage generation and use reusable workflow from everest-ci

### PR DESCRIPTION
**Pull request by SebaLukas**
## Describe your changes
Adding a basic code coverage generation functionality.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

**Comment by SebaLukas** [comment]
By using catch2, some errors are still displayed during code coverage generation. Either we adapt the readme so that the errors are ignored or we try to get the errors out.

**Comment by SebaLukas** [comment]
@andistorm can you have a look into the failing `Deploy html coverage report ` job. I think there is an access problem with the compiler account?

File: CMakeLists.txt [hunk 1]
```diff
@@ -54,9 +56,26 @@ add_subdirectory(input)
...
 if (ISO15118_BUILD_TESTING)
+    set(CMAKE_BUILD_TYPE Debug CACHE STRING "Build type" FORCE) # TODO(sl): Check if necessary
+
     include(CTest)
     list(APPEND CMAKE_MODULE_PATH ${CPM_PACKAGE_catch2_SOURCE_DIR}/extras)
     add_subdirectory(test)
+
+    # TODO(sl): Maybe move to test folder
+    set(CODE_COVERAGE_VERBOSE ON)
+    set(GCOVR_ADDITIONAL_ARGS "--gcov-ignore-errors=no_working_dir_found")
```
Comment (andistorm) [inline CMakeLists.txt:67]: is this with gcovr 8 still necessary?

File: CMakeLists.txt [hunk 2]
```diff
@@ -54,9 +56,26 @@ add_subdirectory(input)
 add_subdirectory(src)
 
 if (ISO15118_BUILD_TESTING)
+    set(CMAKE_BUILD_TYPE Debug CACHE STRING "Build type" FORCE) # TODO(sl): Check if necessary
+
     include(CTest)
     list(APPEND CMAKE_MODULE_PATH ${CPM_PACKAGE_catch2_SOURCE_DIR}/extras)
     add_subdirectory(test)
+
+    # TODO(sl): Maybe move to test folder
```
Comment (andistorm) [inline CMakeLists.txt:65]: Indeed I would move this to test dir

File: CMakeLists.txt (hunk 1 above)
Comment (SebaLukas) [reply CMakeLists.txt:67]: Without this the CI jobs does not work anymore. So I added this again for the CI

File: CMakeLists.txt (hunk 2 above)
Comment (SebaLukas) [reply CMakeLists.txt:65]: I will test this if that is possible

**Review by andistorm — APPROVED** [review]
LGTM 🐧

**Review by MarzellT — APPROVED** [review]

**Pull request merged**